dev/ApplicationData/ApplicationData.idl

```diff
@@ -0,0 +1,190 @@
+        String SharedLocalPath { get; };
+
+        /// Return the path for the temporary data store.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.temporaryfolder
+        String TemporaryPath { get; };
+
+        /// Return a StorageFolder for the local cache data store not included in backup and restore operations.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.localcachefolder
+        Windows.Storage.StorageFolder LocalCacheFolder { get; };
+
+        /// Return a StorageFolder for the local data store. This location is backed up to the cloud.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.localfolder
+        Windows.Storage.StorageFolder LocalFolder { get; };
+
+        /// Return a StorageFolder for the machine data store.
+        /// @note This location is per-machine (not per-user like other *Folder properties).
+        Windows.Storage.StorageFolder MachineFolder { get; };
+
+        /// Return a StorageFolder for the roaming data store.
+        /// @warning Roaming data and settings are no longer supported after Windows 10 version 1901 (aka 19H1 aka 10.0.18363.0).
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingfolder
+        Windows.Storage.StorageFolder RoamingFolder { get; };
+
+        /// Return a StorageFolder for the shared data store.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.sharedlocalfolder
+        Windows.Storage.StorageFolder SharedLocalFolder { get; };
+
+        /// Return a StorageFolder for the temporary data store.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.temporaryfolder
+        Windows.Storage.StorageFolder TemporaryFolder { get; };
+
+        /// Return the settings container in the local data store.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.localsettings
+        ApplicationDataContainer LocalSettings { get; };
+
+        /// Return the settings container in the roaming data store.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingsettings
+        ApplicationDataContainer RoamingSettings { get; };
+
+        /// Remove all data from the local, local cache, roaming, and temporary data stores.
+        /// @see LocalCachePath
+        /// @see LocalCacheFolder
+        /// @see LocalPath
+        /// @see LocalFolder
+        /// @see RoamingPath
+        /// @see RoamingFolder
+        /// @see TemporaryPath
+        /// @see TemporaryFolder
+        [method_name("ClearAsync")]
+        Windows.Foundation.IAsyncAction ClearAllAsync();
+
+        /// Remove all data from the specified data store.
+        Windows.Foundation.IAsyncAction ClearAsync(ApplicationDataLocality locality);
```

Comment:
In terms of ABI
* `ClearAllAsync` requests emission as `ClearAsync`
* `ClearAsync` default requests emission as `ClearAsync`

This creates a conflict that is resolved (via midl) as:
* `ClearAllAsync()` becomes `ClearAsync()`
* `ClearAsync(locality)` becomes `ClearAsync2(locality)`

This is not ideal for languages that don't support overloading (e.g., Rust) and seems entirely avoidable with a better naming scheme that more closely aligns with the behavior of the function.

Suggestions:
* Remove `method_name` attribute to keep the `ClearAllAsync` name. The intent--to clear all localities--is better communicated with the All qualifier. Without it, it's unclear which localities are chosen and requires the developer to consult the docs.
* Leave `ClearAsync` as-is

Reply:
Hmmm. Good point. Well, at least it's consistent with the OS API

Thanks for the feedback

Reply:
FYI per API Review we decided to drop ClearAllAsync and ClearMachineAsync and just have ClearAsync(locality).

The Clear-vs-ClearAll was a Win8-era convenience but as ApplicationData's evolved over the years 'All' lost some meaning (All...as-of-10+years-ago...). If we were doing it from scratch today we wouldn't have the overload and gasp! We are doing it from scratch (sorta) :-)

Reply:
Thanks for the heads up, sounds reasonable.